Add Resolve and Thread buttons to Report-command mailbox message.

b-b-blueberry opened the pull request.

b-b-blueberry committed:
Update `report_context` command:

- Add UI modal with text field for optional comments.
- Replace text message in mailbox with formatted embed.

b-b-blueberry committed:
Skip commonbot `send_message` method.

b-b-blueberry committed:
Add Resolve and Thread buttons to Report-command mailbox message.

b-b-blueberry committed:
Merge remote-tracking branch 'aquova/origin/master'

# Conflicts:
#	src/forwarder.py

b-b-blueberry committed:
Remove duplicate code

b-b-blueberry committed:
Update Report-command for finding existing reply threads.

aquova commented on src/context.py:
```diff
@@ -33,15 +102,15 @@ async def on_submit(self, interaction: discord.Interaction):
                 value=field[1] if len(field[1]) < len_max else f"{field[1][:len_max-1]}…",
```
This line throws an error, as `reported_user` doesn't have a `len` function.

b-b-blueberry replied:
Fixed, accidentally replaced a mention with a user instance

aquova replied:
It doesn't look like this has been changed?

b-b-blueberry replied:
This line was never an issue - the problem was I was referencing `reported_user` instead of `reported_user.mention` several lines below, where the `User` object has no `len` method (despite having a `str()` method returning user#discriminator), but the `str` mention obviously does. Feel free to test the changes.

b-b-blueberry replied:
UGH nevermind the changes were lost somewhere along the way

aquova commented on src/context.py:
```diff
@@ -1,7 +1,75 @@
+            log_chan = interaction.guild.get_channel(MAILBOX)
+            thread: discord.Thread = await message_forwarder.get_or_create_user_reply_thread(
+                user=self.reported_user,
+                parent_channel=log_chan
```
This function doesn't have a `parent_channel` parameter?

b-b-blueberry replied:
It did when I committed 🌧

aquova reviewed (changes requested):
Did some basic testing. After changing it so the error on line 102 didn't happen, the Resolve button worked (it just vanished, I think that's the idea), but the Thread button throws a new error, due to line 48.

b-b-blueberry committed:
Fix errors on Report-command:

- Add missing `mention` field to `reported_user` reference in embed fields.
- Make `reported_user` params required on `ReportThreadButton`.
- Fix wrong params sent to `forwarder.get_or_create_user_reply_thread`.

b-b-blueberry committed:
FIX STUPID

b-b-blueberry committed:
Merge remote-tracking branch 'b-b-blueberry/origin/master'

aquova reviewed (approved):
I think this looks good.

I do have a feature request suggestion - It might be nice to make the embed start off as red, then when you press the "Resolve" button, it turns to green + adds the emoji. I'm fine if we don't want to do that though, or if we want to do it in a later addition.

aquova commented:
I was mistaken, it does start off as a different embed color, then turns to green. Disregard.
